dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala

```diff
@@ -0,0 +1,1056 @@
+                                        outputs,
+                                        timeouts.contractTimeout.toUInt32)
+
+    subtractFeeFromOutputs(refundTxNoFee, feeRate, outputs.map(_.scriptPubKey))
+  }
+
+  def createRefundSig(): Future[(Transaction, PartialSignature)] = {
+    val fundingTx = createUnsignedFundingTransaction
+    val refundTx = createUnsignedRefundTx
+
+    val sigF = PSBT
+      .fromUnsignedTx(refundTx)
+      .addUTXOToInput(fundingTx, index = 0)
+      .addScriptWitnessToInput(P2WSHWitnessV0(fundingSPK), index = 0)
+      .sign(inputIndex = 0, fundingPrivKey)
+      .map(_.inputMaps.head.partialSignatures.head)
+
+    sigF.map((refundTx, _))
+  }
+
+  /** Constructs the (time-locked) refund transaction for when the oracle disappears
+    * or signs an unknown message.
+    * Note that both parties have the same refund transaction.
+    */
+  def createRefundTx(
+      remoteSig: PartialSignature): Future[(Transaction, PartialSignature)] = {
+    val psbt = PSBT.fromUnsignedTx(createUnsignedRefundTx)
+
+    val signedPSBTF = psbt
+      .addSignature(remoteSig, inputIndex = 0)
+      .addUTXOToInput(createUnsignedFundingTransaction, index = 0)
+      .addScriptWitnessToInput(P2WSHWitnessV0(fundingSPK), index = 0)
+      .sign(inputIndex = 0, fundingPrivKey)
+
+    signedPSBTF.flatMap { signedPSBT =>
+      val sig = signedPSBT.inputMaps.head.partialSignatures
+        .filter(_.pubKey == fundingPubKey)
+        .head
+
+      val txT = signedPSBT.finalizePSBT.flatMap(_.extractTransactionAndValidate)
+      val txF = Future.fromTry(txT)
+      txF.map((_, sig))
+    }
+  }
+
+  def createCETWin(
+      remoteSig: PartialSignature): (Future[Transaction], P2WSHWitnessV0) = {
+    createCET(
+      sigPubKey = sigPubKeyWin,
+      remoteSig = remoteSig,
+      payout = winPayout,
+      remotePayout = remoteWinPayout
+    )
+  }
+
+  def createCETLose(
+      remoteSig: PartialSignature): (Future[Transaction], P2WSHWitnessV0) = {
+    createCET(
+      sigPubKey = sigPubKeyLose,
+      remoteSig = remoteSig,
+      payout = losePayout,
+      remotePayout = remoteLosePayout
+    )
+  }
+
+  def createCETWinRemote(): Future[
+    (Transaction, PartialSignature, P2WSHWitnessV0)] = {
+    createCETRemote(
+      sigPubKey = sigPubKeyWin,
+      payout = winPayout,
+      remotePayout = remoteWinPayout
+    )
+  }
+
+  def createCETLoseRemote(): Future[
+    (Transaction, PartialSignature, P2WSHWitnessV0)] = {
+    createCETRemote(
+      sigPubKey = sigPubKeyLose,
+      payout = losePayout,
+      remotePayout = remoteLosePayout
+    )
+  }
+
+  def setupDLCAccept(
+      sendSigs: (
+          PartialSignature,
+          PartialSignature,
+          PartialSignature) => Future[Unit],
+      getSigs: Future[
+        (
+            PartialSignature,
+            PartialSignature,
+            PartialSignature,
+            Vector[PartialSignature])]): Future[SetupDLC] = {
```

Comment:
That would be one thing I could do but I imagine in the future (once this part of the code becomes more stable when I've written interaction code) this will all end up in a case class which is why I have it elaborated rn.